### DISCUSSION TITLE
Add another directory of python codes to build as a Android app.

### DISCUSSION THREAD

**Discussion by zhiheng-ldj** [beeware/briefcase#723] [Q&A]
Hi dear,

Thanks for your help!

I am trying to import some python codes in 'my_code', which is located under './helloworld'.

Specifically, it should seems like this:

<img width="368" alt="截屏2022-04-24 18 04 51" src="https://user-images.githubusercontent.com/67734380/164971279-8c007c14-4c22-4b0e-81d9-563bc234c948.png">

However, when I try to build it as .apk, I encounter that:

```
04-24 17:58:52.432  3932  3932 D Python  : Running 'helloworld' as __main__...
04-24 17:58:52.821  3932  3932 E Python  : Application quit abnormally!
04-24 17:58:52.821  3932  3932 E Python  : Traceback (most recent call last):
04-24 17:58:52.821  3932  3932 E Python  :   File "/data/user/0/com.example.helloworld/files/python/stdlib/lib/python3.9/runpy.py", line 197, in _run_module_as_main
04-24 17:58:52.822  3932  3932 E Python  :     return _run_code(code, main_globals, None,
04-24 17:58:52.822  3932  3932 E Python  :   File "/data/user/0/com.example.helloworld/files/python/stdlib/lib/python3.9/runpy.py", line 87, in _run_code
04-24 17:58:52.822  3932  3932 E Python  :     exec(code, run_globals)
04-24 17:58:52.822  3932  3932 E Python  :   File "/data/user/0/com.example.helloworld/files/python/user_code/app/helloworld/__main__.py", line 1, in <module>
04-24 17:58:52.822  3932  3932 E Python  :     from helloworld.app import main
04-24 17:58:52.822  3932  3932 E Python  :   File "/data/user/0/com.example.helloworld/files/python/user_code/app/helloworld/app.py", line 8, in <module>
04-24 17:58:52.823  3932  3932 E Python  :     import my_code
04-24 17:58:52.823  3932  3932 E Python  : ModuleNotFoundError: No module named 'my_code'
......
```

So my question is :
```
how can I build my_code dir into my app? 
```

Thanks for your help!
Best regards!

**Comment by freakboy3742** [comment]
Briefcase only includes source files named in the `sources` attribute of your `pyproject.toml`. By default, this includes `src/<appname>`; but you can add any folders you want to be included in your app bundle.

So - if you set `sources` to:
```
sources = ['src/helloworld', 'my_code']
```
you should be able to `import my_code` inside your app.

**Reply by zhiheng-ldj** [reply]
Hi @freakboy3742 ,

It works!

By the way, I encountered that:

```
ModuleNotFoundError: No module named 'cv2'
```

Can I use my python env inside my app?

Thanks!

**Reply by freakboy3742** [reply]
The app that is packaged by Briefcase is completely isolated from your development virtual environment. That's how we can guarantee that you can distribute your app to someone else (who won't have the same virtual environment configured).

If you have a dependency on a third-party library, you should add that library to the `requires` list in your `pyproject.toml`. The `requires` declaration takes the same syntax as pip (so, you can install specific versions, version ranges, etc).
